ElementX/Sources/Screens/StartChat/StartChatViewModel.swift

```diff
@@ -37,8 +42,80 @@ class StartChatViewModel: StartChatViewModelType, StartChatViewModelProtocol {
         case .createRoom:
             callback?(.createRoom)
         case .inviteFriends:
-            // TODO: start invite people flow
             break
+        case .selectUser(let user):
+            showLoadingIndicator()
+            Task {
+                let currentDirectRoom = await userSession.clientProxy.directRoomForUserIdentifier(user.userId)
+                switch currentDirectRoom {
+                case .success(.some(let roomId)):
+                    self.hideLoadingIndicator()
+                    self.callback?(.openRoom(withIdentifier: roomId))
+                case .success(nil):
+                    await self.createDirectRoom(with: user)
+                case .failure(let failure):
+                    self.hideLoadingIndicator()
+                    self.displayError(failure)
+                }
+            }
+        }
+    }
+    
+    func displayError(_ type: ClientProxyError) {
+        switch type {
+        case .failedRetrievingDirectRoom:
+            state.bindings.alertInfo = AlertInfo(id: type,
+                                                 title: ElementL10n.dialogTitleError,
+                                                 message: ElementL10n.retrievingDirectRoomError)
+        case .failedCreatingRoom:
+            state.bindings.alertInfo = AlertInfo(id: type,
+                                                 title: ElementL10n.dialogTitleError,
+                                                 message: ElementL10n.retrievingDirectRoomError)
+        default:
+            state.bindings.alertInfo = AlertInfo(id: type)
         }
     }
+    
+    // MARK: - Private
+    
+    private func setupBindings() {
+        context.$viewState
+            .map(\.bindings.searchQuery)
+            .debounce(for: .milliseconds(300), scheduler: DispatchQueue.main)
+            .sink(receiveValue: { [weak self] searchQuery in
+                if MatrixEntityRegex.isMatrixUserIdentifier(searchQuery) {
+                    self?.state.searchResults = [UserProfileProxy(userId: searchQuery, displayName: nil, avatarURL: nil)]
+                } else {
+                    self?.state.searchResults = []
+                }
+            })
```

Comment:
Can we prefer trailing closures please?
```suggestion
            .sink { [weak self] searchQuery in
                if MatrixEntityRegex.isMatrixUserIdentifier(searchQuery) {
                    self?.state.searchResults = [UserProfileProxy(userId: searchQuery, displayName: nil, avatarURL: nil)]
                } else {
                    self?.state.searchResults = []
                }
            }
```